### PR TITLE
arch/imxrt/gpt: add timer interface

### PR DESCRIPTION
**Pull request by sunghan-chang**
This is mandatory to use HW timer in user context.

Signed-off-by: sunghan-chang <sh924.chang@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [cd72f93a9f66a16d8d6a4773bc6b012e2abacb13] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cd72f93a9f66a16d8d6a4773bc6b012e2abacb13] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [81dd95a391bd7f631f72a806d2d2d1ceb19d5684] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [81dd95a391bd7f631f72a806d2d2d1ceb19d5684] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [e1b218670721f420a7e141da2a689f1579f6de76] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [e1b218670721f420a7e141da2a689f1579f6de76] - Code Rule Check (C++) OK.

File: os/arch/arm/src/imxrt/imxrt_timer_lowerhalf.c [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+	if (lower->gpt == NULL) {
+		return -EINVAL;
+	}
+
+	/*
+	 * Register the timer driver as /dev/timerX.  The returned value from
+	 * timer_register is a handle that could be used with timer_unregister().
+	 * REVISIT: The returned handle is discard here.
+	 */
+	if (!timer_register(devpath, (struct timer_lowerhalf_s *)lower)) {
```
Comment (Taejun-Kwon) [inline os/arch/arm/src/imxrt/imxrt_timer_lowerhalf.c]: This is lowerhalf of timer, so it should be registered to Timer as a interface of lower device controller.
So timer_register should be called in board code, I think imxrt_boot.c is acceptable.
Board code should init config first, and then init lowerhalf with config, finally timer.
I don't know about timer properly, but it seems that below should be applied.
```
struct imxrt_gpt_lowerhalf_s *gpt
gpt = imxrt_gpt_configure();
if (gpt == NULL) {
   error;
}
timer_lowerhalf_s *timer = imxrt_timer_initialize(gpt );
if (timer == NULL) {
  error;
}
timer_register(devpath, timer);
....
```

**Comment by seinfra** [comment]
Target : [5f069266141b1e5e223fc7b3ce0c54756b25837e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5f069266141b1e5e223fc7b3ce0c54756b25837e] - Code Rule Check OK.

**Pull request merged**